akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/CommandHandler.scala

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+package akka.persistence.typed.javadsl
+
+import akka.actor.typed.javadsl.ActorContext
+import akka.annotation.InternalApi
+import akka.japi.pf.FI
+import akka.persistence.typed.scaladsl.PersistentBehaviors.EffectImpl
+import akka.util.OptionVal
+
+/**
+ * SAM for reacting on commands
+ *
+ * Used with [[CommandHandlerBuilder]] to setup the behavior of a [[PersistentBehavior]]
+ */
+@FunctionalInterface
+trait CommandHandler[Command, Event, State] {
+  def apply(ctx: ActorContext[Command], state: State, command: Command): Effect[Event, State]
+}
+/**
+ * SAM for reacting on signals
+ *
+ * Used with [[CommandHandlerBuilder]] to setup the behavior of a [[PersistentBehavior]]
+ */
+@FunctionalInterface
+trait CommandToEffect[Command, MsgCommand <: Command, Event, State] {
```

Comment:
I think the type bound may go up in smoke in Java and become `Object` when it refers to another type parameter.

Reply:
:boom:

I can get rid of this extra one now we've got the abstract class, i think

Reply:
This also appears to work in java, you end up with 

e.g. this won't compile

``` static CommandToEffect<Command, String, Event, EventsInFlight>```

Where String does not extend Command